### PR TITLE
Improve exptrigsimp()

### LINKED ISSUE #10124: Simplification fails to recognize sin expressed as exponentials
```

In [2]: exp(Matrix([[0, -1, 0], [1, 0, 0], [0, 0, 0]]))
Out[2]: 
⎡    -ⅈ    ⅈ          -ⅈ      ⅈ   ⎤
⎢   ℯ     ℯ        ⅈ⋅ℯ     ⅈ⋅ℯ    ⎥
⎢   ─── + ──     - ───── + ────  0⎥
⎢    2    2          2      2     ⎥
⎢                                 ⎥
⎢     ⅈ      -ⅈ      -ⅈ    ⅈ      ⎥
⎢  ⅈ⋅ℯ    ⅈ⋅ℯ       ℯ     ℯ       ⎥
⎢- ──── + ─────     ─── + ──     0⎥
⎢   2       2        2    2       ⎥
⎢                                 ⎥
⎣      0               0         1⎦

In [3]: simplify(_)
Out[3]: 
⎡     cos(1)       -sin(1)  0⎤
⎢                            ⎥
⎢  ⎛     2⋅ⅈ⎞  -ⅈ            ⎥
⎢ⅈ⋅⎝1 - ℯ   ⎠⋅ℯ              ⎥
⎢────────────────  cos(1)   0⎥
⎢       2                    ⎥
⎢                            ⎥
⎣       0             0     1⎦

In [4]: m = _

In [5]: fu(_)
Out[5]: 
⎡     cos(1)       -sin(1)  0⎤
⎢                            ⎥
⎢  ⎛     2⋅ⅈ⎞  -ⅈ            ⎥
⎢ⅈ⋅⎝1 - ℯ   ⎠⋅ℯ              ⎥
⎢────────────────  cos(1)   0⎥
⎢       2                    ⎥
⎢                            ⎥
⎣       0             0     1⎦

In [6]: sqrt
sqrt           sqrt_mod       sqrt_mod_iter  sqrtdenest     

In [6]: sqrtdenest(_)
Out[6]: 
⎡    cos(1)      -sin(1)  0⎤
⎢                          ⎥
⎢     ⅈ      -ⅈ            ⎥
⎢  ⅈ⋅ℯ    ⅈ⋅ℯ              ⎥
⎢- ──── + ─────  cos(1)   0⎥
⎢   2       2              ⎥
⎢                          ⎥
⎣      0            0     1⎦

In [7]: trig
trigamma      trigonometry  trigsimp      

In [7]: trigsimp(_)
Out[7]: 
⎡    cos(1)      -sin(1)  0⎤
⎢                          ⎥
⎢     ⅈ      -ⅈ            ⎥
⎢  ⅈ⋅ℯ    ⅈ⋅ℯ              ⎥
⎢- ──── + ─────  cos(1)   0⎥
⎢   2       2              ⎥
⎢                          ⎥
⎣      0            0     1⎦

```

The expression for `sin(1)` has not been recognized, while expressions for `cos` and `-sin(1)` have.

### PR DESCRIPTION
**Pull request by ylemkimon**
- Removed simplify from exptrigsimp as simplify already includes exptrigsimp.
- Improved(refactored) exptrigsimp() algorithm for more flexible detection of hyperbolic functions. Fixes #10124 and comment of #13106.

### CODE REVIEW

**Comment by smichr** [comment]
This needs to be merged with master to resolve conflicts.

**Comment by ylemkimon** [comment]
@smichr Resolved 😄

File: sympy/simplify/tests/test_trigsimp.py [hunk 1]
```diff
@@ -378,12 +380,8 @@ def valid(a, b):
     for a in (1, I, x, I*x, 1 + I):
         w = exp(a)
         eq = y*(w - 1/w)/(w + 1/w)
-        s = simplify(eq)
-        assert s == exptrigsimp(eq)
```
Comment (Upabjojr) [inline sympy/simplify/tests/test_trigsimp.py]: Has this assert been removed?

Comment (ylemkimon) [reply sympy/simplify/tests/test_trigsimp.py]: @Upabjojr As simplify was called in exptrigsimp, exptrigsimp was equivalent to simplify and did nothing more than it, and that test was checking it.

**Comment by Upabjojr** [comment]
OK, thanks for your contribution!

**Pull request merged**